legend-engine-xts-data-space/legend-engine-xt-data-space-grammar/src/main/java/org/finos/legend/engine/language/pure/grammar/from/DataSpaceParseTreeWalker.java

```diff
@@ -257,9 +257,21 @@ private DataSpaceExecutable visitDataSpaceTemplateExecutable(DataSpaceParserGram
         DataSpaceParserGrammar.ExecutableDescriptionContext descriptionContext = PureGrammarParserUtility.validateAndExtractOptionalField(ctx.executableDescription(), "description", executable.sourceInformation);
         executable.description = descriptionContext != null ? PureGrammarParserUtility.fromGrammarString(descriptionContext.STRING().getText(), true) : null;
 
-        // query
+        // query could either be type of PackageableElementPointer or Lambda
         DataSpaceParserGrammar.ExecutableTemplateQueryContext queryContext = PureGrammarParserUtility.validateAndExtractRequiredField(ctx.executableTemplateQuery(), "query", executable.sourceInformation);
-        executable.query = visitLambda(queryContext.combinedExpression());
+        if (queryContext.functionIdentifier() != null)
+        {
+            DataSpaceParserGrammar.ExecutableTemplateQueryContext functionContext = PureGrammarParserUtility.validateAndExtractRequiredField(ctx.executableTemplateQuery(), "query", executable.sourceInformation);
+            executable.query = new PackageableElementPointer(
+                    PackageableElementType.FUNCTION,
+                    functionContext.functionIdentifier().getText(),
+                    walkerSourceInformation.getSourceInformation(functionContext.functionIdentifier())
+            );
+        }
+        else if (queryContext.combinedExpression() != null)
+        {
+           executable.query = visitLambda(queryContext.combinedExpression());
+        }
```

Comment:
What happens if `queryContext.functionIdentifier()` and `queryContext.combinedExpression()` are both null?

Reply:
Thanks! I throw an exception now for unsupported query type